abi/src/main/java/org/web3j/abi/TypeDecoder.java

```diff
@@ -490,7 +490,7 @@ private static <T extends Type> T decodeDynamicStructElements(
                                         0,
                                         TypeReference.create(declaredField));
                         staticOffset +=
-                                staticStructNestedPublicFieldsFlatList((Class<Type>) classType)
+                                staticStructNestedPublicFieldsFlatList((Class<Type>) declaredField)
```

Comment:
Could you please add a unit test to cover this issue?

Reply:
Sure. The new test cases have been added to both EncoderTest and DecoderTest.